external/s3-sqs/src/main/scala/org/apache/spark/sql/sqs/SqsClient.scala

```diff
@@ -0,0 +1,267 @@
+          logDebug("Successfully parsed sqs message")
+          list :+ ((path, timestampMills, messageReceiptHandle))
+        } else {
+          if (eventName.contains("ObjectRemoved")) {
+            if (!ignoreFileDeletion) {
+              exception = Some(new SparkException("ObjectDelete message detected in SQS"))
+            } else {
+              logInfo("Ignoring file deletion message since ignoreFileDeletion is true")
+            }
+          } else {
+            logWarning("Ignoring unexpected message detected in SQS")
+          }
+          errorMessages.append(messageReceiptHandle)
+          list
+        }
+      } catch {
+        case me: MappingException =>
+          errorMessages.append(message.getReceiptHandle)
+          logWarning(s"Error in parsing SQS message ${me.getMessage}")
+          list
+        case e: Exception =>
+          errorMessages.append(message.getReceiptHandle)
+          logWarning(s"Unexpected error while parsing SQS message ${e.getMessage}")
+          list
+      }
+    }
+    if (errorMessages.nonEmpty) {
+      addToDeleteMessageQueue(errorMessages.toList)
+    }
+    parsedMessages
+  }
+
+  private def convertTimestampToMills(timestamp: String): Long = {
+    val timeInMillis = timestampFormat.parse(timestamp).getTime()
+    timeInMillis
+  }
+
+  private def evaluateRetries(): Unit = {
+    retriesOnFailure += 1
+    if (retriesOnFailure >= sqsMaxRetries) {
+      logError("Max retries reached")
+      exception = Some(new SparkException("Unable to receive Messages from SQS for " +
+        s"${sqsMaxRetries} times Giving up. Check logs for details."))
+    } else {
+      logWarning(s"Attempt ${retriesOnFailure}." +
+        s"Will reattempt after ${sqsFetchIntervalSeconds} seconds")
+    }
+  }
+
+  private def createSqsClient(): AmazonSQS = {
+    try {
+      val isClusterOnEc2Role = hadoopConf.getBoolean(
+        "fs.s3.isClusterOnEc2Role", false) || hadoopConf.getBoolean(
+        "fs.s3n.isClusterOnEc2Role", false) || sourceOptions.useInstanceProfileCredentials
+      if (!isClusterOnEc2Role) {
+        val accessKey = hadoopConf.getTrimmed("fs.s3n.awsAccessKeyId")
+        val secretAccessKey = new String(hadoopConf.getPassword("fs.s3n.awsSecretAccessKey")).trim
+        logInfo("Using credentials from keys provided")
+        val basicAwsCredentialsProvider = new BasicAWSCredentialsProvider(
+          accessKey, secretAccessKey)
+        AmazonSQSClientBuilder
+          .standard()
+          .withClientConfiguration(new ClientConfiguration().withMaxConnections(maxConnections))
+          .withCredentials(basicAwsCredentialsProvider)
+          .withRegion(region)
+          .build()
+      } else {
+        logInfo("Using the credentials attached to the instance")
+        val instanceProfileCredentialsProvider = new InstanceProfileCredentialsProviderWithRetries()
+        AmazonSQSClientBuilder
```

Comment:
See: https://github.com/steveloughran/hadoop/blob/trunk/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java#L1206

proxy support is essential; you can grab the keys from org.apache.hadoop.fs.s3a.Constants; things like timeout and buffer size a matter of preference

Reply:
yeah thanks for pointing out, will pick this